From 68eca5e94e6f63f4392b15581ad41af52e0edfd9 Mon Sep 17 00:00:00 2001
From: Robin Watts <robin.watts@artifex.com>
Date: Thu, 6 Sep 2018 12:56:51 +0100
Subject: [PATCH] Fix SEGV seen in all-devices test with plank
 examples/ridt91.eps

Silly typo in the contone fill code for the landscape flippedy
color cache case.
---
 base/gximono.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/base/gximono.c b/base/gximono.c
index 2971f40..276db69 100644
--- a/base/gximono.c
+++ b/base/gximono.c
@@ -1222,8 +1222,8 @@ image_render_mono_ht(gx_image_enum * penum_orig, const byte * buffer, int data_x
                                 dev_value = &(color_cache[*psrc * spp_out]);
                                 for (j = 0; j < spp_out; j++) {
                                     *(devc_contone[j] + position) = dev_value[j];
-                                    position -= LAND_BITS;
                                 }
+                                position -= LAND_BITS;
                                 xr--;
                             }           /* at loop exit xn will be <= xr */
                             psrc++;
@@ -1259,8 +1259,8 @@ image_render_mono_ht(gx_image_enum * penum_orig, const byte * buffer, int data_x
                                 dev_value = &(color_cache[*psrc * spp_out]);
                                 for (j = 0; j < spp_out; j++) {
                                     *(devc_contone[j] + position) = dev_value[j];
-                                    devc_contone[j] += LAND_BITS;
                                 }
+                                position += LAND_BITS;
                                 xr++;
                             }           /* at loop exit xn will be >= xr */
                             psrc++;
-- 
2.9.1


